### PR TITLE
Remove toggling of unique names in subscenes

### PR DESCRIPTION
**Pull request by SaracenOne**
After some discussions, simply disallowing the ability to turn off the unique_name node flag in subscenes seems much more appropiate given how the system actually works. The unique name system works in the scope node ownership, so it does indeed make sense to disallow toggling it on the subscenes.

Deprecates previous attempt of fixing it here #82545

### CODE REVIEW

**Comment by timothyqiu** [comment]
* Toggle option in the right click menu should also be removed when appropriate.
* For the unique name button in the tree, I think it's better to disable it instead of reporting an error when clicked.

**Comment by SaracenOne** [comment]
Good thinking. Fixed! Though for the first issue you raised, the editor already does this, the ability to toggle the unique name off via the icon being a janky incongruity

File: editor/gui/scene_tree_editor.cpp [hunk 1]
```diff
@@ -169,12 +169,17 @@ void SceneTreeEditor::_cell_button_pressed(Object *p_item, int p_column, int p_i
 		NodeDock::get_singleton()->get_parent()->call("set_current_tab", tab_container->get_tab_idx_from_control(NodeDock::get_singleton()));
 		NodeDock::get_singleton()->show_groups();
 	} else if (p_id == BUTTON_UNIQUE) {
-		undo_redo->create_action(TTR("Disable Scene Unique Name"));
-		undo_redo->add_do_method(n, "set_unique_name_in_owner", false);
-		undo_redo->add_undo_method(n, "set_unique_name_in_owner", true);
-		undo_redo->add_do_method(this, "_update_tree");
-		undo_redo->add_undo_method(this, "_update_tree");
-		undo_redo->commit_action();
+		if (n->get_owner() == EditorNode::get_singleton()->get_edited_scene()) {
```
Comment (KoBeWi) [inline editor/gui/scene_tree_editor.cpp:172]: Why this change if the button is disabled and can't be clicked?

**Comment by KoBeWi** [comment]
It's still possible to toggle unique name using shortcut.

**Comment by SaracenOne** [comment]
Good catch, I forgot about the shortcut since I don't think it's assigned by default. I've added the correct handling to the tool_selected case now. It will now simply disallow any toggling to unowned nodes. It will also only pop up the error dialog if ALL the nodes in the selection are unowned by the edited scene. I think this is the right call since it will still allow big batch converts on the entire scene even if a couple of editable children get caught in the selection.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
